module/mailer/src/Infrastructure/Sender/Strategy/SymfonyMailerStrategy.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\Mailer\Infrastructure\Sender\Strategy;
+
+use Ergonode\Mailer\Domain\MailInterface;
+use Ergonode\Mailer\Infrastructure\Sender\MailerStrategyInterface;
+use Symfony\Bridge\Twig\Mime\TemplatedEmail;
+use Symfony\Component\Mailer\MailerInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ */
+class SymfonyMailerStrategy implements MailerStrategyInterface
+{
+    /**
+     * @var MailerInterface
+     */
+    private MailerInterface $mailer;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private TranslatorInterface $translator;
+
+    /**
+     * @var string|null
+     */
+    private ?string $defaultFrom;
+
+    /**
+     * @var string|null
+     */
+    private ?string $defaultReplyTo;
+
+    /**
+     * @param MailerInterface     $mailer
+     * @param TranslatorInterface $translator
+     * @param string|null         $defaultFrom
+     * @param string|null         $defaultReplyTo
+     */
+    public function __construct(
+        MailerInterface $mailer,
+        TranslatorInterface $translator,
+        ?string $defaultFrom = null,
+        ?string $defaultReplyTo = null
+    ) {
+        $this->mailer = $mailer;
+        $this->translator = $translator;
+        $this->defaultFrom = $defaultFrom;
+        $this->defaultReplyTo = $defaultReplyTo;
+    }
+
+    /**
+     * @see We need to wait until Symfony resolve https://github.com/symfony/symfony/issues/35925
+     *
+     * @param MailInterface $mail
+     *
+     * @throws \Throwable
+     */
+    public function send(MailInterface $mail): void
+    {
+        $previousLocale = $this->translator->getLocale();
+        $this->translator->setLocale($mail->getTemplate()->getLanguage()->getLanguageCode());
+
+        try {
+            $email = (new TemplatedEmail())
+                ->addTo(...$mail->getRecipient()->getTo()->asStringArray())
```

Comment:
Should i use Demeter Law (https://en.wikipedia.org/wiki/Law_of_Demeter)?

Reply:
Meaning you'd shorten the usage to sth like `$mail->getWithJustOneMethod()`? If yes personally I'd avoid doing it as it increases the overall complexity of classes